base/flat_set.h

```diff
@@ -0,0 +1,100 @@
+#ifndef SLINKY_BASE_FLAT_SET_H
+#define SLINKY_BASE_FLAT_SET_H
+
+#include <algorithm>
+#include <cassert>
+#include <iterator>
+#include <vector>
+
+namespace slinky {
+
+// Implements a set with std::vector<T> storage. The vector is kept sorted to enable efficient set operations.
+template <typename T>
+class flat_set {
```

Comment:
What is the benefit of this over std::set? The insert and erase both have linear time instead of logarithmic.

Reply:
That it can be turned into a span for use with `depends_on`. However, at this point, I'm not actually doing that any more, I  will just remove this.